### DISCUSSION TITLE
Bug: Hard to migrate to flat configs with default Firebase Functions.

### DISCUSSION THREAD

**Discussion by MauScheff** [eslint/eslint#18679] [Q&A] [answered]
### Environment

Node version: 20
npm version: 10.7.0
Local ESLint version: 9.3
Global ESLint version: 9.3
Operating System: Linux codespaces-788de8 6.5.0-1022-azure #23~22.04.1-Ubuntu SMP Thu May  9 17:59:24 UTC 2024 x86_64 GNU/Linux

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Try this</summary>

```
1. Follow this steps to install Firebase https://firebase.google.com/docs/functions/get-started?gen=2nd
2. Init a new project.
3. Change eslint to 9.3
4. Run the migration asssitant
5. Find bugs. In my case: 
```
</details>

```js
 Key "rules": Key "valid-jsdoc": Could not find "valid-jsdoc" in plugin "@".
```

### What did you expect to happen?

As a "default" user, I would like that when I run the migration, the new config file just works.

### What actually happened?

I ran into many bugs, and gave up on upgrading. In my workflow I can't afford to spend a lot of time getting ESLint to work by fixing all problems.

### Link to Minimal Reproducible Example

https://firebase.google.com/docs/functions/get-started?gen=2nd

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by aladdin-add** [comment]
>  Key "rules": Key "valid-jsdoc": Could not find "valid-jsdoc" in plugin "@".

The require-jsdoc and valid-jsdoc rules have been removed in ESLint v9.0.0. These rules were initially deprecated in 2018.

To address: Use the [replacement rules](https://github.com/gajus/eslint-plugin-jsdoc/wiki/Comparison-with-deprecated-JSdoc-related-ESLint-rules) in eslint-plugin-jsdoc.

please read the migration docs before upgrading to v9: https://eslint.org/docs/latest/use/migrate-to-9.0.0#-removed-require-jsdoc-and-valid-jsdoc-rules

**Comment by ingophmeyer** [comment]
I also get this "TypeError: Key "rules": Key "valid-jsdoc": Could not find "valid-jsdoc" in plugin "@".". I even have a new project and never ever used it. There is no reference anywhere. Searching for jsdoc gives 0 results in my project. It is really frustrating. Nobody I know is able to run eslint 9. I hate that so much. Outdated documentation everywhere. I also give up.

**Comment by emaysyuk** [comment]
In case somebody don't need valid-jsdoc and require-jsdoc, here's a valid config for ESLint 9.19.0 that can be used with Cloud Functions:
https://stackoverflow.com/questions/78989693/eslint-9-x-x-error-typescript-eslint-parser/79413403#79413403

**Comment by hugovdm** [comment]
Solving by trial-and-error commenting-out of rules, it seems the "google" ruleset was the one that was invoking valid-jsdoc:

    extends: fixupConfigRules(
      compat.extends(
        'eslint:recommended',
        'plugin:import/errors',
        'plugin:import/warnings',
        'plugin:import/typescript',
        // 'google',
        'plugin:@typescript-eslint/recommended',
      ),
    ),

(I had also started with "firebase init functions".)

**Comment by hugovdm** [comment, marked as answer]
I suppose this is probably the ticket tracking the incompatibility issue:

https://github.com/google/eslint-config-google/issues/75
